gwpy/types/series.py

```diff
@@ -978,19 +978,41 @@ def crop(self, start=None, end=None, copy=False):
                  % type(self).__name__)
             end = None
 
+        # check if series is irregular
+        try:
+            _ = self.dx
+            irregular = False
+        except AttributeError:
+            irregular = True
```

Comment:
There isn't a need to assign the `self.dx` to a variable in this case:

 ```suggestion
        try:
            self.dx
        except AttributeError:
            irregular = True
        else:
            irregular = False
```